radicle-node/src/worker.rs

```diff
@@ -191,15 +192,22 @@ impl<G: Signer + Ecdh + 'static> Worker<G> {
                 // a state we can't roll back.
                 cmd.arg("--prune");
             }
+            Namespaces::Many(_) => {
+                // Same case as All
+            }
         }
 
         if self.atomic {
             // Enable atomic fetch. Only works with Git 2.31 and later.
             cmd.arg("--atomic");
         }
+
+        // Ignore our own remote when fetching
+        let mut fetchspecs = fetch.namespaces.as_fetchspecs();
+        fetchspecs.push(format!("^refs/namespaces/{}/*", self.local));
```

Comment:
How about adding a `Namespaces::Not(PubKey)` variant to help here?

Reply:
It _feels_ out of place, since some of the logic using `Namespaces` cares about whom it's fetching from. For example, whether to add the `--prune` flag.

I could add another struct though. Something like:
```rust
pub struct Exclude(NonEmpty<PublicKey>);

impl Exclude {
    pub fn as_fetchspecs(&self) -> Vec<String> {
        /* code here */
    }
}
```

Reply:
Hmm true, maybe something to think about for `git-ref-format` at some point.